src/stack.rs

```diff
@@ -0,0 +1,285 @@
+//! A safe stack.
+//!
+//! The `Stack` allows inserting, removing and iterating it's elements.
+//!
+//! NOTE: This was written for a learning purpose.
+
```

Comment:
If you'd be up for it, you could implement the from trait for the linked list: linkedlist -> stack so to speak